miri-script/src/commands.rs

```diff
@@ -0,0 +1,711 @@
+use itertools::chain;
+use std::collections::BTreeMap;
+
+use std::io::Result;
+use std::path::{Path, PathBuf};
+
+use duct::cmd;
+use path_macro::path;
+use walkdir::WalkDir;
+
+use crate::arg::Subcommands;
+
+const JOSH_FILTER: &str =
+    ":rev(75dd959a3a40eb5b4574f8d2e23aa6efbeb33573:prefix=src/tools/miri):/src/tools/miri";
+
+fn detect_miri_dir() -> std::io::Result<PathBuf> {
+    const MIRI_SCRIPT_ROOT_DIR: &str = env!("CARGO_MANIFEST_DIR");
```

Comment:
Should this be `const`? I feel like we want this at runtime, no?

Reply:
I don't think we support putting this binary anywhere else, so compile-time seems fine, no?

Reply:
This program looks like our version of `x`. So I think a user would be justified in assuming that they should `cargo install` it then be able to cd around or run this in multiple clones or a worktree just fine.

But it seems like it is not our version of `x`, so it would be nice to either prevent that entirely (can we make `cargo install` an error? I don't think so?) or if it's not that hard, handle it.

Reply:
The entrypoint is still `./miri`, right? So users shouldn't ever even see this crate unless they go look for it. We can have a comment in the Cargo.toml saying that this is not meant to be installed.

Reply:
I don't know what the entrypoint is meant to be :p

So long as we have a comment or something about this situation, I'm willing to wait if/until someone files an issue because they tried to `cargo install` this crate.

Reply:
> I don't know what the entrypoint is meant to be :p

Well we certainly don't want `cargo run --manifest-path=...`.

OTOH, how is that `./miri` wrapper supposed to be able to know where the manifest is? That needs the readdir logic and an issue with exactly that logic is what triggered the rewrite in the first place... maybe `$(dirname "$0")/miri-script/Cargo.toml` is good enough (but then I wonder why we had that normalization logic in there in the first place :shrug: )

Reply:
Do we have any consensus here? I'd personally lean towards detecting Miri dir at compile time.